torch/utils/_sympy/value_ranges.py

```diff
@@ -401,28 +412,46 @@ def is_integer(val):
 
         a = ValueRanges.wrap(a)
         b = ValueRanges.wrap(b)
-        if a.is_singleton() and b.is_singleton():
-            r = a.lower**b.lower
+        # Not implemented yet. It's a bit tricky
+        # If you want to implement it, compute the partial derivatives of a ** b
+        # and check the ranges where the function is increasing / decreasing
+        # Another non-tight way of doing this is defaulting to doing noting that for a > 0,  a ** b == exp(b * log(a))
+        # If this second option is implemented, by carefult about the types and possible infinities here and there.
+        if not b.is_singleton():
+            return ValueRanges.unknown()
+
+        b = b.lower
+        if a.is_singleton():
+            a = a.lower
+            r = a ** b
             if r == sympy.zoo:
                 return ValueRanges.unknown()
             return ValueRanges.wrap(r)
-        elif b.is_singleton() and is_integer(b.lower) and b.lower >= 0:
-            i = ValueRanges.wrap(1)
-            for _ in range(b.lower):
-                i = cls.mul(i, a)
-            return i
-        else:
-            # This is fairly difficult to analyze, so give up for anything
-            # complicated
-            return ValueRanges.unknown()
 
-    @staticmethod
-    def minimum(a, b):
-        return ValueRangeAnalysis.min_or_max(a, b, sympy.Min)
+        if not is_integer(b):
+            if b < 0 or a.lower < 0:
+                return ValueRanges.unknown()
+            return ValueRanges.increasing_map(a, lambda x: x ** b)
+
+        # exponentiation by squaring
+        if b < 0:
+            a = cls.reciprocal(a)
+            b = -b
+        acc = ValueRanges.wrap(1)
+        while b > 0:
+            if b % 2 == 1:
+                acc = cls.mul(acc, a)
+            a = cls.mul(a, a)
+            b = b // 2
```

Comment:
Can we not just call into sympy.pow at this point ? Feel like there's probably a way of implementing this simpler

Reply:
Yeah, let me rewrite this. This one's a weird op, as `-3.0 ** x` is not well defined, but `-3 ** x` is well defined for x integer. I'll split it into `a` integer and not.

Reply:
Ok, I heavily simplified this logic, now it looks much better.